### PR TITLE
Closes #479

### LINKED ISSUE #479: Decode gzip/deflate automatically when mocking a proxied request
**Is your feature request related to a problem? Please describe.**
When the origin server that I'm proxying responses with gzip/deflate encoded body, mockoon just shows the raw content when mocking the request. This is annoying since it's impossible to quickly adjust the response in the mock. See https://github.com/mockoon/mockoon/discussions/453

**Describe the solution you'd like**

- mockoon should automatically decode these response bodies when creating a mock
- the returned mocked response should then be encoded again
- an icon could indicate that the original response was and and the mocked one will be gzipped
- optionally there could be a way of switching back to the raw content, but I personally don't have a use case for that

**Describe alternatives you've considered**

- there also could be a checkbox in the proxy settings to disable this feature, but I would prefer the icon/toggle described above

**Additional context**

image

I worked around it by setting a proxy request header, but that's not very intuitiv:
image

### PR DESCRIPTION
**Pull request by rubaiate**
createRouteFromLog: update encoding headers
  Indicate response content encoding in ui

**Parent issue** 

Closes #479

**Technical implementation details**

**Checklist**

- [ ] automated tests added
- [ ] data migration added (if applicable)
- [ ] data migration automated tests added (if applicable)

### CODE REVIEW

**Comment by 255kb** [comment]
Thanks for opening the PR @rubaiate 
Also, you don't need to close a PR if you want to make changes, you can push a new commit on your branch and the PR will be updated.

**Comment by 255kb** [comment]
I will probably review this in two weeks. Next week I will have limited access to my computer, so don't worry if you don't hear from me in the coming days :)

**Comment by barefootdeveloper** [comment]
This is going to be a really significant feature, thank you for taking it on! 

I have one concern about the current implementation. I believe if we create a mocked response off of proxied request with a compressed response, it would still create an invalid response. This is because the headers stay, showing that the response is compressed and encoded.

The alternatives I see are:

1) When decoding the response, alter the headers as well so it becomes a valid response.
2) Move the decoding to the UI so that you can see the decoded response. Add support for compressing responses.

The first option is easier, but results in a different endpoint behavior from the original. I believe this is the approach that @255kb suggested.

The second option seems the most ideal since it maintains the original API behavior, but I suspect most callers of an API would handle this transparently so it might be mostly an academic argument.

File: src/renderer/app/services/environments.service.ts [hunk 1]
```diff
@@ -474,11 +474,26 @@ export class EnvironmentsService {
       if (log.response) {
         const headers: Header[] = [];
         log.response.headers.forEach((header) => {
+          if (
+            header.key === 'content-encoding' ||
```
Comment (barefootdeveloper) [inline src/renderer/app/services/environments.service.ts:478]: I think we may need transfer-encoding added to this list to facilitate the mock this endpoint.

Comment (rubaiate) [reply src/renderer/app/services/environments.service.ts:478]: As of now, transfer-encoding is not checked also while decoding messages. Maybe need to check at both decoding and header update. I'll check what is possible.
Thanks.

**Comment by 255kb** [comment]
> This is going to be a really significant feature, thank you for taking it on!
> 
> I have one concern about the current implementation. I believe if we create a mocked response off of proxied request with a compressed response, it would still create an invalid response. This is because the headers stay, showing that the response is compressed and encoded.
> 
> The alternatives I see are:
> 
> 1. When decoding the response, alter the headers as well so it becomes a valid response.
> 2. Move the decoding to the UI so that you can see the decoded response. Add support for compressing responses.
> 
> The first option is easier, but results in a different endpoint behavior from the original. I believe this is the approach that @255kb suggested.
> 
> The second option seems the most ideal since it maintains the original API behavior, but I suspect most callers of an API would handle this transparently so it might be mostly an academic argument.

Indeed, I supported the version 1, where you can create a mock from compressed content while still being able to modify it. Also, I don't see the automatic mock creation as a way to recreate 1:1 what has been received by mockoon but rather to have a base to create a new mock. 
If we want to support encoding of outgoing responses, it should be an activable option at the route or env level, and part of another feature request.

**Comment by 255kb** [comment]
> This is going to be a really significant feature, thank you for taking it on!
> 
> I have one concern about the current implementation. I believe if we create a mocked response off of proxied request with a compressed response, it would still create an invalid response. This is because the headers stay, showing that the response is compressed and encoded.
> 
> The alternatives I see are:
> 
> 1. When decoding the response, alter the headers as well so it becomes a valid response.
> 2. Move the decoding to the UI so that you can see the decoded response. Add support for compressing responses.
> 
> The first option is easier, but results in a different endpoint behavior from the original. I believe this is the approach that @255kb suggested.
> 
> The second option seems the most ideal since it maintains the original API behavior, but I suspect most callers of an API would handle this transparently so it might be mostly an academic argument.

Indeed, I supported the version 1, where you can create a mock from compressed content while still being able to modify it. Also, I don't see the automatic mock creation as a way to recreate 1:1 what has been received by mockoon but rather to have a base to create a new mock. 
If we want to support encoding of outgoing responses, it should be an activable option at the route or env level, and part of another feature request.

**Comment by 255kb** [comment]
@rubaiate Let me know if you want me to take it from here, finalize and add tests.

**Comment by rubaiate** [comment]
Great. 👌

**Pull request merged**